### PR TITLE
Ensures "Exit app" link contains an exit icon

### PR DESCRIPTION
**Pull request by tonytlwu**
Ref. https://github.com/Fliplet/fliplet-studio/issues/4065

Requires https://github.com/Fliplet/fliplet-api/pull/3231

### CODE REVIEW

File: js/menu.js [hunk 1]
```diff
@@ -2,13 +2,36 @@ var $menuElement = $('[data-name="Bottom icon bar"]');
...
+        '</div>',
+      '</li>'
+    ].join(''));
+    $li.on('click', function onExitClick() {
+      Fliplet.Navigate.exitApp();
+    });
+    $(this).find('ul').append($li);
+
+    // Prevent default "Exit app" link from being added
+    return Promise.reject('');
```
Comment (squallstar) [inline js/menu.js]: @tonytlwu you should reject with a truth-y error

File: js/menu.js [hunk 2]
```diff
@@ -29,4 +52,8 @@ function init() {
       action: 'about-overlay'
     });
   });
+}
+
+if (menuInstanceId) {
```
Comment (squallstar) [inline js/menu.js]: not sure why this has been moved down here, feels like an unnecessary change.

File: js/menu.js (hunk 1 above)
Comment (tonytlwu) [reply js/menu.js]: @squallstar Is that the convention? I followed the Form Builder method of rejecting with an empty string to stop further actions. I believe that's because Form Builder uses the reject message as error message.

File: js/menu.js (hunk 2 above)
Comment (tonytlwu) [reply js/menu.js]: It is unnecessary. It's moved to follow our standard _variables-functions-then-code_ flow. That's all.

File: js/menu.js (hunk 1 above)
Comment (squallstar) [reply js/menu.js]: @tonytlwu that was just a hack to avoid the form builder from displaying an error messages, yes. Ideally I would prefer rejecting with either undefined (so no arguments) or a specific error like `new Error('ManuallyHandled')`, e.g. something we can parse and understand.

Comment (squallstar) [reply js/menu.js]: As a further note, FormBuilder should be updated not to attempt displaying an error when undefined is given. We initially sent an empty string so it wouldn't fail when parsing it.

Comment (tonytlwu) [reply js/menu.js]: OK. I'll reject with no arguments then. I'll leave the form builder changes to you when you next work on the component.
